compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -205,6 +293,90 @@ private void addPlatformLibraryPaths(PackageContext packageContext,
         }
     }
 
+    private void addUsedPlatformLibraryPaths(PackageContext packageContext,
+                                         PlatformLibraryScope scope,
+                                         Set<JarLibrary> libraryPaths, HashMap<PackageId, HashSet<String>> pkgWiseUsedNativeClassPaths) {
+        // Add all the jar library dependencies of current package (packageId)
+        Collection<PlatformLibrary> otherJarDependencies = jBalBackend.platformLibraryDependencies(
+                packageContext.packageId(), scope);
+
+        HashSet<String> usedNativeClassPaths = pkgWiseUsedNativeClassPaths.get(packageContext.packageId());
+
+        for (PlatformLibrary otherJarDependency : otherJarDependencies) {
+            JarLibrary newEntry = (JarLibrary) otherJarDependency;
+
+            if (otherJarDependencies.size() < 2 && !isUsedDependency(newEntry, usedNativeClassPaths)) {
+                continue;
+            }
+
+            if (newEntry.groupId().isEmpty() || newEntry.artifactId().isEmpty() || newEntry.version().isEmpty()) {
+                libraryPaths.add(new JarLibrary(otherJarDependency.path(), scope, getPackageName(packageContext)));
+                continue;
+            }
```

Comment:
We can have a separate method to check these conditions instead of having multiple `if` statements

Reply:
Fixed in ca4f4407bec9daec6edb965d3f0ad7eb8b206669.